src/Nethermind/Nethermind.Api/IEngineRequestsTracker.cs

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Api;
+
+public interface IEngineRequestsTracker
+{
+    void OnForkchoiceUpdatedCalled();
+    void OnNewPayloadCalled();
+}
+
+public class NullEngineRequestsTracker : IEngineRequestsTracker
+{
+    public static readonly NullEngineRequestsTracker Instance = new();
+
```

Comment:
Add a private parameterless constructor to prevent future missuse:

```suggestion
    private NullEngineRequestsTracker() { }
    public static readonly NullEngineRequestsTracker Instance = new();
```

Reply:
This change might be irrelevant if we can remove the class altogether.